### DISCUSSION TITLE
Looking to exclude tests from inputs

### DISCUSSION THREAD

**Discussion by HaveSpacesuit** [vercel/turborepo#6979] [Help] [answered]
I am finding that my cache is hitting when I add this as my build inputs:

```json
"pipeline": {
  "build": {
    "dependsOn": ["^build"],
    "inputs": ["!**/*.test.*"],
    "outputs": ["esm/**", "cjs/**", "dist/**"]
  }
```

My intention is to keep the cache valid if I change a test. However, this setup is keeping cached builds on _any_ file change. If I remove the `inputs`, then the build re-runs appropriately.

**Comment by mehulkar** [comment, marked as answer]
If you set `inputs`, it opts out of all inclusions also. So you need to specify `src/**` (or wherever your files are). You can 👍🏾 this issue for this feature: https://github.com/vercel/turbo/issues/6814

**Reply by HaveSpacesuit** [reply]
Thanks. I guess I would have expected this to be taking anything under the workspace, except for the test files, since all source files should match that glob.
